spec/controllers/packages_controller_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.describe PackagesController, type: :controller do
+
+  describe "GET 'rates'" do
+    it "is successful" do
+      get :rates, :state => "GA", :city => "Atlanta", :zip => 30307
+      expect(response.response_code).to eq 200
+    end
+
+    it "returns json" do
+      get :rates, :state => "GA", :city => "Atlanta", :zip => 30307
+      expect(response.header['Content-Type']).to include 'application/json'
+    end
+
+    context "the returned json object" do
+      before :each do
+        get :rates, :state => "GA", :city => "Atlanta", :zip => 30307
+        @response = JSON.parse response.body
+      end
+
+      it "is an array of shipping objects" do
+        expect(@response).to be_an_instance_of Hash
+        expect(@response.length).to eq 2
+      end
+
+      it "includes the appropriate keys" do
+        expect(@response.keys).to eq ["ups", "usps"]
+      end
+    end
+
+    context "ups response" do
+      it "returns an array of prices" do
```

Comment:
When naming your tests, you want to be as accurate as possible (to properly reflect what you're testing). Though nit picky, it'd be more accurate to say `it "returns an array"` since you're not actually checking that the array contains prices.